packages/blitz-paint/src/text.rs

```diff
@@ -87,3 +87,31 @@ pub(crate) fn stroke_text<'a>(
         }
     }
 }
+
+/// Draw selection highlight rectangles for the given byte range in a layout.
+/// Uses Parley's Selection type for accurate geometry calculation.
+pub(crate) fn draw_text_selection(
+    scene: &mut impl PaintScene,
+    layout: &Layout<TextBrush>,
+    transform: Affine,
+    selection_start: usize,
+    selection_end: usize,
+) {
+    let anchor = Cursor::from_byte_index(layout, selection_start, Affinity::Downstream);
+    let focus = Cursor::from_byte_index(layout, selection_end, Affinity::Downstream);
+    let selection = Selection::new(anchor, focus);
+
+    let selection_color = color::palette::css::STEEL_BLUE;
+    let scale = layout.scale() as f64;
+
+    selection.geometry_with(layout, |rect, _line_idx| {
+        // Convert from Parley's BoundingBox to kurbo::Rect, accounting for scale
+        let rect = kurbo::Rect::new(
+            rect.x0 as f64 / scale,
+            rect.y0 as f64 / scale,
+            rect.x1 as f64 / scale,
+            rect.y1 as f64 / scale,
```

Comment:
These `/ scale` are incorrect. Parley dimensions are in physical pixels, and painting code draws in physical pixels. So you can just remove the conversion entirely.